libflux/flux-core/src/formatter/tests.rs

```diff
@@ -1169,3 +1175,40 @@ test2 = 2",
 test2 = 2",
     );
 }
+#[test]
+fn preserve_multiline_test() {
+    // ensure functions given preserve their structure
+    //assert_unchanged("test _convariance_missing_column_2 = () =>
+    //({input: testing.loadStorage(csv: inData), want: testing.loadMem(csv: outData), fn: covariance_missing_column_2})");
+
+    assert_unchanged(
+        "test _covariance_missing_column_2 = () => ({
+    input: testing.loadStorage(csv: inData),
+    want: testing.loadMem(csv: outData),
+    fn: covariance_missing_column_2,
+})",
+    );
+
+    assert_unchanged(
+        r#"event = (
+        url,
+        username,
+        password,
+        action="EventsRouter",
+        methods="add_event",
+        type="rpc",
+        tid=1,
+        summary="",
+        device="",
+        component="",
+        severity,
+        eventClass="",
+        eventClassKey="",
+        collector="",
+        message="") => {
+    body = json.encode(v: payload)
+
+    return http.post(headers: headers, url: url, data: body)
+}"#,
```

Comment:
This is great. Do we have any tests that explicitly check that single line funcs are changed to this format? Is that something easy to test?

Reply:
I don't think we have an assert format but by assessing the changes in the other flux files I think we can conclude that it works.
<img width="729" alt="Screen Shot 2021-06-01 at 8 16 12 PM" src="https://user-images.githubusercontent.com/53279816/120418897-3886fe80-c316-11eb-8d89-a816bd48bf0c.png">

Reply:
@BarWeiner1 It would be helpful to have the kind of test that faith is talking about, if we don't already. That way we can be sure that we don't accidentally regress this behavior with future formatter changes.

Reply:
Yeah I'd like to see some tests added here: 1) that checks that a function with 4 or more params gets reformatted and 2) that checks that a function with 3 or fewer params does not get reformatted.